lotto645.py

```
@@ -63,98 +67,123 @@ def buy_lotto645(
-        tlmt_date = soup.find("input", id="WAMT_PAY_TLMT_END_DT").get('value')
+        soup = BS(html, "html5lib")
+        
+        try:
+            draw_date_el = soup.find("input", id="ROUND_DRAW_DATE")
+            tlmt_date_el = soup.find("input", id="WAMT_PAY_TLMT_END_DT")
+            
+            if draw_date_el and tlmt_date_el:
+                draw_date = draw_date_el.get('value')
+                tlmt_date = tlmt_date_el.get('value')
+            else:
+                raise ValueError("Date inputs not found")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)